### ISSUE TITLE
Use for IPFS

### ISSUE THREAD

**Issue by jbenet** [davidar/vector-web#2]
- [x] welcoming landing page #1 
- [x] deploy clientside on IPFS 
- [x] keep list of deployed versions
- [x] pin all code
- [x] make "IPFS Chatroom" buttons for repos
- [ ] add to all the repos

**Comment by davidar** [comment]
v0: https://ipfs.io/ipfs/QmXkxZvZ8apvsVL7naAwsL8kGxu41PSBHzvTpaw39oowBB

v1: https://ipfs.io/ipfs/QmUTJhudRGn2QRUz8eJ7wn8wxvpaJa5eBmhEexcwdPNa7h

v2: https://ipfs.io/ipfs/Qmbm8matKreFd1nDwA3CWWCMauMcSEJNz1NVJ4ApEzJbhp

Should probably set up IPNS for this: [![IPFS Chat](https://img.shields.io/badge/Chat-Live-brightgreen.svg)](https://ipfs.io/ipfs/QmUTJhudRGn2QRUz8eJ7wn8wxvpaJa5eBmhEexcwdPNa7h/#/room/#freenode_#ipfs:matrix.org)

**Comment by davidar** [comment]
@jbenet also, it would be cool if ipfs.io could run a [Matrix homeserver](https://github.com/matrix-org/synapse) and an official [IRC bridge](https://github.com/matrix-org/matrix-appservice-irc) for #ipfs, just so that this doesn't go offline if something happens to matrix.org

@lgierth this would also allow us to extend the auth mechanism to handle PKI challenges like you suggested (I think that was you?)

**Comment by jbenet** [comment]
@davidar that's great!! 

i like v2 the most, but i like the v1 logo (like the overlap of the cube and V). (the text overlap is not ideal though, i think). maybe: "IPFS [ overlapped-logo ] Vector"

Yeah we could run a homeserver and irc bridge. easy to setup and run? maybe can add to https://github.com/ipfs/infrastructure as a PR and discuss there how best to set it up.

on the "Chat Live" badge, i think a brighter badge/button may be merited. i mostly ignore those buttons. i think something with a gif with chat text scrolling may help. idk thought cc @diasdavid

**Comment by davidar** [comment]
@jbenet  yeah, I like the v1 logo too, but @ara4n asked me not to use the vector branding

I haven't setup a matrix server myself, but I believe it's relatively easy. I'll open an issue

**Comment by daviddias** [comment]
> i mostly ignore those buttons. i think something with a gif with chat text scrolling may help.

scrolling text ftw! :D 

I've been adding this [![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs) to every IPFS related repo I create, it is simple, but it goes nice with the other badges

**Comment by jbenet** [comment]
@davidar great work! wonder: can we modify the client so it automatically joins a channel (in our case, #ipfs on freenode) or something? after signing up i was lost. i know you mentioned it in the splash landing page (which looks great!) but dont think it's enough. people will be lost in a new thing, joining new channels, (the matrix default channel) and start asking questions there. it will be a nightmare for matrix users and the chat users both.

**Comment by davidar** [comment]
> can we modify the client so it automatically joins a channel (in our case, #ipfs on freenode)

@dbkr Would you mind helping me with this? :)

**Comment by dbkr** [comment]
This is fairly high on our todo list anyway so you can wait until we write it if you like, but essentially it's just a question of extending the URL handling to remember if the client was opened with a room URL and then redirect back to it after signup.

**Comment by davidar** [comment]
> This is fairly high on our todo list anyway so you can wait until we write it if you like

Awesome. @jbenet That ok with you?

**Comment by ara4n** [comment]
We also plan to fix the anonymous login stuff shortly too, which I think is jbenet's main concern here.

**Comment by jbenet** [comment]
> Awesome. @jbenet That ok with you?

Sure!

@dbkr @ara4n do you guys need styling help for the vector UI/UX? we may be able to help with that.

**Comment by dbkr** [comment]
I've just landed support for redirecting back to the right room after logging in to the develop branch of vector: https://github.com/vector-im/vector-web/tree/develop

Hopefully this should do the trick - let me know if you have any problems with it.

**Comment by davidar** [comment]
@dbkr Sorry, I must have missed your comment. I'll give it a spin soon and let you know how it goes :)

**Comment by davidar** [comment]
@jbenet http://chat.ipfs.io/#/room/#ipfs:matrix.org

**Comment by jbenet** [comment]
nice!! :+1:
